### ISSUE TITLE
0.32.5: java.lang.Exception: Invalid response from database driver. No :status provided.

### ISSUE THREAD

**Issue by lkraav** [metabase/metabase#9818]
Loading a dashboard with multiple questions, a couple of these show up in the log:

```
04-20 11:57:58 DEBUG middleware.log :: POST /api/card/83/query 200 [ASYNC: completed] 737 ms (13 DB calls) Jetty threads: 8/50 (4 busy, 2 idle, 0 queued) (63 total active threads)                                                          
04-20 11:57:58 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.                                        
java.lang.Exception: Invalid response from database driver. No :status provided.true                                                                                                                                                         
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)                                                                                                     
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)                                                                          
        at metabase.query_processor.middleware.cache$run_query_with_cache$respond__34325.invoke(cache.clj:110)                                                                                                                               
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)                                                                         
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)                                                                                   
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)                                                                                                                                             
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)                                                                                                                                                   
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)                        
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)                                                                                                                                           
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)                                                                    '                                                                             
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)                                                                                                                                      
        at clojure.lang.AFn.run(AFn.java:22)                                                                                                                                                                                                 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)                                                 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)                                                                                                                                                   
        at java.lang.Thread.run(Thread.java:748)                                                                                
04-20 11:57:58 INFO middleware.cache :: Query took 678 ms to run; miminum for cache eligibility is 10000 ms                                                                                                                                   
04-20 11:57:58 INFO middleware.cache :: Query took 683 ms to run; miminum for cache eligibility is 10000 ms                                                                                                                                   
04-20 11:57:58 DEBUG async.api-response :: Async response finished, closing channels.                                                                                                                                                         
04-20 11:57:58 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.                                                                                                                            
java.lang.Exception: Invalid response from database driver. No :status provided.true                                                                                                                                                          
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97) 
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)                                                                                                     
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)                                                                            
        at metabase.query_processor.middleware.cache$run_query_with_cache$respond__34325.invoke(cache.clj:110)                                                                                                                               
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)                                                                                   
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)                                           
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)                                                                                                                                                   
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)                   
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)                                                                                                                                           
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)                                 
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)                                                                                                                                      
        at clojure.lang.AFn.run(AFn.java:22)                                                                                    
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)                                                                                                                                                   
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)                                                               '                                                                                    
        at java.lang.Thread.run(Thread.java:748)
```

All questions contents seem to have loaded successfully this time, whereas it's been usual for some dashboard question card to display a database connection error, where the solution is to try reloading the dashboard until all questions display their results.

Searching "provided.true" here had 0 results, so posting this.

**Comment by camsaul** [comment]
Hi @lkraav, the changes I made in 0.32.5 are the reason you don't need to reload the Dashboard a few times to get everything to show up. It sounds like it's working correctly.

Regarding the error messages:
1. There's supposed to be a space after `provided.`. There's no `provided.true` anywhere
2.  The error message is a case of logging being a little more verbose than it needs to be. If a query gets canceled or things like that it some of the middleware that's supposed to process query results will get confused and log error messages when there are no results to transform. You can ignore those for now. I will work on cleaning up the false warnings for the next release.

**Comment by eschultze** [comment]
FYI I'm getting the same.
```
04-25 18:01:33 DEBUG async.api-response :: Async response finished, closing channels.
04-25 18:01:33 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
04-25 18:01:33 DEBUG async.api-response :: Async response finished, closing channels.
04-25 18:01:33 DEBUG middleware.log :: POST /api/card/52/query 200 [ASYNC: completed] 255 ms (15 DB calls) Jetty threads: 8/50 (2 busy, 5 idle, 0 queued) (65 total active threads)
04-25 18:01:33 DEBUG async.api-response :: Async response finished, closing channels.
04-25 18:01:33 DEBUG middleware.log :: POST /api/card/51/query 200 [ASYNC: completed] 235 ms (15 DB calls) Jetty threads: 8/50 (2 busy, 5 idle, 0 queued) (65 total active threads)
04-25 18:01:33 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
04-25 18:02:44 DEBUG middleware.log :: GET /api/dashboard/3 200 32 ms (9 DB calls) Jetty threads: 8/50 (3 busy, 6 idle, 0 queued) (71 total active threads)
04-25 18:02:45 DEBUG async.api-response :: starting streaming response
04-25 18:02:45 DEBUG async.semaphore-channel :: Permit available without waiting, will run fn immediately
04-25 18:02:45 DEBUG async.util :: Running metabase.query_processor.middleware.async$async__GT_sync$fn__33067@19b0648b on separate thread...
04-25 18:02:45 DEBUG async.util :: Running metabase.query_processor.middleware.async$async__GT_sync$fn__33067@19b0648b on separate thread...
04-25 18:02:45 DEBUG async.api-response :: starting streaming response
04-25 18:02:46 DEBUG async.api-response :: Response not ready, writing one byte & sleeping...
04-25 18:02:46 DEBUG async.api-response :: Response not ready, writing one byte & sleeping...
04-25 18:02:47 DEBUG async.api-response :: Async response finished, closing channels.
04-25 18:02:47 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
04-25 18:02:47 DEBUG middleware.log :: POST /api/card/8/query 200 [ASYNC: completed] 2 s (16 DB calls) Jetty threads: 8/50 (2 busy, 6 idle, 0 queued) (75 total active threads)
04-25 18:02:47 WARN middleware.async :: Unhandled exception, exepected `catch-exceptions` middleware to handle it.
java.lang.Exception: Invalid response from database driver. No :status provided.true
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invokeStatic(process_userland_query.clj:97)
        at metabase.query_processor.middleware.process_userland_query$format_userland_query_result.invoke(process_userland_query.clj:87)
        at clojure.core$partial$fn__5828.invoke(core.clj:2638)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623$fn__33625.invoke(async_wait.clj:49)
        at metabase.query_processor.middleware.async_wait$wait_for_permit$fn__33563$fn__33602$state_machine__8574__auto____33623.invoke(async_wait.clj:49)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:973)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:972)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:977)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:975)
        at clojure.core.async.impl.ioc_macros$take_BANG_$fn__8592.invoke(ioc_macros.clj:986)
        at clojure.core.async.impl.channels.ManyToManyChannel$fn__3478$fn__3479.invoke(channels.clj:95)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
04-25 18:02:47 DEBUG middleware.log :: POST /api/card/15/query 200 [ASYNC: completed] 2 s (15 DB calls) Jetty threads: 8/50 (2 busy, 6 idle, 0 queued) (75 total active threads)
04-25 18:02:47 DEBUG async.api-response :: Async response finished, closing channels.
```

**Comment by camsaul** [comment]
Fixed by #9911
